File: tourcount/src/main/java/com/wmstein/tourcount/widgets/EditNotesWidget.java
package com.wmstein.tourcount.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.wmstein.tourcount.R;

import java.util.Objects;

/**********************************
 * EditNotesWidget used by CountOptionsActivity
 * Created by wmstein on 2016-02-18,
 * last edited on 2020-09-19
 */
public class EditNotesWidget extends LinearLayout
{
    private final TextView widget_title;
    private final EditText notes_name;

    public EditNotesWidget(Context context, AttributeSet attrs)
    {
        super(context, attrs);

        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_title, this, true);
        widget_title = findViewById(R.id.widgetTitle);
        notes_name = findViewById(R.id.sectionName);
    }

    public void setWidgetTitle(String title)
    {
        widget_title.setText(title);
    }

    public void setNotesName(String name)
    {
        notes_name.setText(name);
    }

    public void setHint(String hint)
    {
        notes_name.setHint(hint);
    }

    public String getNotesName()
    {
        return notes_name.getText().toString();
    }

}

File: tourcount/src/main/java/com/wmstein/tourcount/widgets/EditNotesWidget.kt
package com.wmstein.tourcount.widgets

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import com.wmstein.tourcount.R
import java.util.Objects

/**********************************
 * EditNotesWidget used by CountOptionsActivity
 * Created by wmstein on 2016-02-18,
 * last edited in Java on 2020-09-19,
 * converted to Kotlin on 2023-07-05
 */
class EditNotesWidget(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {
    private val widget_title: TextView
    private val notes_name: EditText

    init {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_title, this, true)
        widget_title = findViewById(R.id.widgetTitle)
        notes_name = findViewById(R.id.sectionName)
    }

    fun setWidgetTitle(title: String?) {
        widget_title.text = title
    }

    fun setHint(hint: String?) {
        notes_name.hint = hint
    }

    var notesName: String?
        get() = notes_name.text.toString()
        set(name) {
            notes_name.setText(name)
        }
}